### PR TITLE
Add cpp code

### PR DESCRIPTION
**Pull request by molpopgen**
1. Add the code from my repo
2. Add a README for this repo

### CODE REVIEW

File: fwdpy11_arg_example/TODO.md [hunk 1]
```diff
@@ -0,0 +1,32 @@
+# To get msprime integration working:
```
Comment (jeromekelleher) [inline fwdpy11_arg_example/TODO.md]: Maybe drop this file?

File: fwdpy11_arg_example/README.rst [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+
+.. code-block:: bash
+
+    conda create --name ftprime_ms python
+    source activate ftprime_ms
+    conda config --add channels defaults
+    conda config --add channels bioconda
+    conda config --add channels conda-forge
+    conda install gcc fwdpy11==0.1.4 msprime==0.5.0 pybind11==2.2.1 pandas
+    git clone https://github.com/molpopgen/fwdpy11_arg_example
```
Comment (jeromekelleher) [inline fwdpy11_arg_example/README.rst]: Delete next two lines, right?

Comment (jeromekelleher) [reply fwdpy11_arg_example/README.rst]: Or at least, the clone and 'cd' are now redundant. Could say at the start to make sure CWD is where this readme is.

**Review by jeromekelleher — APPROVED** [review]
Works for me!

**Comment by petrelharp** [comment]
Looks good!  We should perhaps explicitly clarify that the example actually **does** need msprime 0.5.0 to run, and some minor changes would be needed to run with the current version: I just tested it with a more recent version and got
```
    trees.dump_tables(nodes=self.__nodes, edges=self.__edges)
TypeError: dump_tables() got an unexpected keyword argument 'nodes'
```

How about
```
**Note:** This code uses msprime version 0.5.0 (installed in the example below using conda). To make it work with a more recent msprime version, some minor changes would be needed.
```

**Comment by molpopgen** [comment]
Done.

**Comment by ashander** [comment]
Looks good to me. Thanks Kevin!

**Review by ashander — APPROVED** [review]

**Comment by petrelharp** [comment]
Rad; thanks.

**Pull request merged**